lib/private/Files/Storage/Local.php

```diff
@@ -372,6 +387,11 @@ public function rename($source, $target): bool {
 		}
 
 		if (@rename($this->getSourcePath($source), $this->getSourcePath($target))) {
+			if ($this->caseInsensitive) {
+				if (strtolower($target) === strtolower($source) && !$this->file_exists($target)) {
```

Comment:
I think we should care to multibyte strings:
```suggestion
				if (mb_strtolower($target) === mb_strtolower($source) && !$this->file_exists($target)) {
```
I renamed a file `É` to `é` and I had the issue.